flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageMemoryManagerImpl.java

```diff
@@ -142,6 +155,8 @@ public void setup(BufferPool bufferPool, List<TieredStorageMemorySpec> storageMe
                     !tieredMemorySpecs.containsKey(memorySpec.getOwner()),
                     "Duplicated memory spec.");
             tieredMemorySpecs.put(memorySpec.getOwner(), memorySpec);
+            numReservedBuffersForDefinitelyRecycledOwners +=
+                    memorySpec.isDefinitelyRecycled() ? memorySpec.getNumGuaranteedBuffers() : 0;
```

Comment:
Shouldn't we ensure capacity upon setting up?

Reply:
I don't think we have to, ensuring capacity before requesting non-reclaimable buffers is enough. Besides,
1. A memory manager without non-guaranteed reclaimable buffers doesn't need to ensure capacity at all.
2. Ensuring capacity upon setting up needs to acquire dozens of buffers at once, Ensuring capacity at runtime may be more smooth.